libs/ml/include/ml/optimisation/optimiser.hpp

```diff
@@ -478,6 +480,18 @@ typename Optimiser<T>::SizeType Optimiser<T>::UpdateBatchSize(SizeType const &ba
   return updated_batch_size;
 }
 
+template <typename T>
+void Optimiser<T>::AddExternalGradientsToGraph(std::vector<TensorType> grads)
```

Comment:
- does it matter that they're "external" - to me it makes sense if this function is called "AddGradientsToGraph".
- Add function comment (explain typical use case of injecting external gradients)